homeassistant/components/zimi/controller.py

```diff
@@ -0,0 +1,113 @@
+"""Zimi Controller wrapper class device."""
+
+import logging
+import pprint
+
+from zcc import (
+    ControlPoint,
+    ControlPointDescription,
+    ControlPointDiscoveryService,
+    ControlPointError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN, PLATFORMS, TIMEOUT, VERBOSITY, WATCHDOG
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ZimiController:
```

Comment:
The zcc helper library is based on a root class being the ControlPoint which acts as a hub for connections to all support devices that are connected via the local zimi mesh.

This class in HA mirrors the zcc class instance and is key for the connect() method which actually uses the ZIMI API to connect and enumerate devices on the mesh.

I set this up, and have tested, multiple zimi cloud controllers on the same HA.

Reply:
I'm not saying this class doesn’t work. Again, since this is a hub, I'd suggest you refactor your code to use a `DataUpdateCoordinator`, move logic like connecting there and remove the things that are just mirroring functionality.

Reply:
Please see https://github.com/home-assistant/core/commit/5d4badf11afc8416a88995aa9c65f8f05f649d74 which introduces the DataUpdateCoordinator.   

Note that the zcc library is not a polled library but rather provides a push update capability via subscribe() / unsubscribe() at the device level.   The ZimiLight class uses this capabliity to receive status updates.

Tested add, remove and operation of lights OK